### ISSUE TITLE
Rule proposal: a-implies-b

### ISSUE THREAD

**Issue by BridgeAR** [sindresorhus/eslint-plugin-unicorn#2655]
### Description

When comparing strict equality of two entries and that another condition must be true for one of the conditions, I often see that both conditions are checked individually again. Instead, recommending to just do that once would be great to minimize code and prevent the CPU cycles calculating the extra equality.

### Examples

Fail ❌

```js
// In any order
a > 0 && b > 0 && a === b

a > 0 && b > 2 && a === b

a < c && b < c && a === b

a !== 'test' && b !== 'test' && a === b

a === 'test' && b === 'test' && a === b
```

Pass ✅

```js
a > 0 && a === b

// a > 0 && b > 2 && a === b
b > 2 && a === b

a < c && a === b

a !== 'test' && a === b

a === 'test' && a === b
```

### Proposed rule name

a-implies-b

### Additional Info

This would not be auto-fixable, since objects could for example have a toPrimitive symbol attached that has a dynamic outcome.

**Comment by fregante** [comment]
This is generic boolean algebra and it could cover a variety of situations. Biome has something similar: https://biomejs.dev/linter/rules/use-simplified-logic-expression/

I also found this: https://github.com/azat-io/eslint-plugin-de-morgan

As for the specific example I think it's too niche
